crates/ra_assists/src/assist_context.rs

```diff
@@ -162,6 +173,30 @@ impl Assists {
         Some(())
     }
 
+    fn add_impl_multiple_files(
+        &mut self,
+        label: Assist,
+        f: impl FnOnce(&mut AssistDirector),
+    ) -> Option<()> {
+        if !self.resolve {
+            return None;
+        }
+        let mut director = AssistDirector::default();
+        f(&mut director);
+        let changes = director.finish();
+        let file_edits: Vec<SourceFileEdit> =
+            changes.into_iter().map(|mut change| change.source_file_edits.pop().unwrap()).collect();
```

Comment:
I'm sure I'm missing something, but is `change.source_file_edits` supposed to have exactly one element?

Reply:
Yes, by design `AssistBuilder` has only edit (see its `finish` method). `AssistDirector` has only one builder per file. But maybe this is not really obvious and may lead to errors if the `AssistBuilder` design changes. Should we assert that we expect only one edit per builder?

Reply:
No, I guess the `unwrap()` is enough then.